plugins/SegmentEditor/javascripts/Segmentation.js

```diff
@@ -93,31 +83,39 @@ Segmentation = (function($) {
         });
 
         segmentation.prototype.markComparedSegments = function() {
-            var comparisonService = window.CoreHome.ComparisonsStoreInstance;
-            var comparedSegments = comparisonService.getSegmentComparisons().map(function (comparison) {
+            const comparisonService = window.CoreHome.ComparisonsStoreInstance;
+            const comparedSegments = comparisonService.getSegmentComparisons().map(function (comparison) {
                 return comparison.params.segment;
             });
-            $('div.segmentList ul li[data-definition]', this.target).removeClass('comparedSegment').filter(function () {
-                var definition = $(this).attr('data-definition');
-                return comparedSegments.indexOf(definition) !== -1 || comparedSegments.indexOf(decodeURIComponent(definition)) !== -1;
-            }).each(function () {
-                $(this).addClass('comparedSegment');
+            $('div.segmentList ul li[data-definition]', this.target).each(function () {
+                const $segment = $(this);
+                const definition = $segment.attr('data-definition');
+                const isCompared = (
+                  comparedSegments.indexOf(definition) !== -1 ||
+                  comparedSegments.indexOf(decodeURIComponent(definition)) !== -1
+                );
+                $segment.toggleClass('comparedSegment', isCompared);
+                $segment.find('.compareSegment').attr('data-state', isCompared ? 'active' : '');
             });
             self.checkIfComparedSegmentsHasReachedLimit();
         };
         segmentation.prototype.checkIfComparedSegmentsHasReachedLimit = function() {
             const limit = piwik.config.data_comparison_segment_limit + 1;
             const comparisonService = window.CoreHome.ComparisonsStoreInstance;
             const comparedSegmentsLength = comparisonService.getSegmentComparisons().length;
-            $('div.segmentList ul li[data-definition] span.compareSegment').each(function() {
+            $('div.segmentList ul li[data-definition] .compareSegment').each(function() {
+              const $compareButton = $(this);
+              const $segment = $compareButton.parent();
+              const currentState = $compareButton.attr('data-state');
+              if (currentState === 'active') {
+                return;
+              }
               if (comparedSegmentsLength >= limit) {
-                $(this).addClass('no-click');
-                $(this).parent().attr('title', _pk_translate('General_MaximumNumberOfSegmentsComparedIs', [limit]));
+                $compareButton.attr('data-state','disabled');
+                addTooltip($compareButton, _pk_translate('General_MaximumNumberOfSegmentsComparedIs', [limit]));
```

Comment:
ℹ️ Add a disabled state for the compare button
ℹ️ The tooltip was on the `li`but it should have been on the `.compareSegment` button

<img width="531" height="415" alt="Image" src="https://github.com/user-attachments/assets/a9097118-4d43-490c-ab5e-b0b9eb08e609" />